File: Models/Category.cs
namespace financial_manager.Models
{
    public class Category
    {
        public int Id { get; init; }
        public string Title { get; set; } = string.Empty;
        public DateTime CreatedAt { get; set; } = DateTime.Now;
        public List<Transaction>? Transactions { get; set; }
        public User? User { get; set; }

        public Category() { }
    }
}

File: financialManagerUI/src/app/entities/models/category.ts
import { BaseEntity } from '../shared/baseEntity';
import { Transaction } from './transaction';
import { User } from './user';

export class Category extends BaseEntity {
  public title: string;
  public createdAt: Date = new Date();
  public user?: User = undefined;
  public transactions?: Transaction[] = undefined;

  constructor(
    title: string,
    user?: User,
    transactions?: Transaction[],
    id?: number
  ) {
    super();
    this.title = title;
    this.user = user;
    this.transactions = transactions;
    this.id = id;
  }
}
